src/Services/Page.vala

```diff
@@ -219,18 +219,55 @@ public class ENotes.PageTable : DatabaseTable {
         }
     }
 
-    private const string[] SYMBOLS_BLACKLIST = {"#", "`", "\t", "<br>", ">", "<", "\n", "~" };
+    // Describs member of BlackList used in cleanup. reg is the regular expression and replace the string it will be replaced
+    private struct BLMember {
+        public GLib.Regex reg;
+        public string replace;
+        
+        public BLMember (GLib.Regex reg, string replace) {
+            this.reg = reg;
+            this.replace = replace;
+        }
+    
+    }
+      
     private string cleanup (string line) {
+ 
+    // This list is used for complex commands mostly by plugins e.g. "" <youtube [Link]>
+       BLMember[] Regex_complex_commands;
+       
+    // @translator this code summarises a notebook page. Instead of given youtube link this code changes into Youtube Video      
+    // [a-zA-Z0-9_\.\?\/:\=\+&\-'"]* is very greedy way for stating website but closer solutions need more space 
+       BLMember youtube_video = BLMember(/<youtube [a-zA-Z0-9_\.\?\/:\=\+&\-'"]*>/, _("Youtube Video"));
+    // "
+    // Explaination for link: Regex for [Something](Something). As greedy as editor on markdown
+       BLMember link = BLMember(/\[[a-zA-Z0-9_\.\?\/:\=\+&\-'"]*\]\([a-zA-Z0-9_\.\?\/:\=\+&\-'"]*\)/, _("Link"));
+       
+    //  \[\^\d+\]:? leads to e.g. [^32], [^68]:   
+       Regex_complex_commands = {BLMember (/<break>/, ""), BLMember (/<highlight>/, ""), BLMember (/<color #[\da-zA-Z]{6}>/, ""), BLMember (/\[\^\d+\]:?/, ""), youtube_video, link};
+       
+    // Regex_Simple_elements used for symbols. Some symbols are part of more complex commands so these
+    // list is used at the end
+    // first regular expression replaces # ~ ` etc. with ""
+        BLMember[] Regex_Simple_elements = {BLMember (/[#\n\t<>]+/, ""), BLMember(/<br>/, "")};
+       
         string output = line;
 
-        if (line.contains ("---")) return "";
-
-        foreach (string item in SYMBOLS_BLACKLIST) {
-            if (output.contains (item)) {
-                output = output.replace (item, "");
+        try {
+            foreach (BLMember item in Regex_complex_commands) {
+                  output = item.reg.replace (output, -1, 0, item.replace);        
+            }  
+        
+            foreach (BLMember item in Regex_Simple_elements) {
+                output = item.reg.replace (output, -1, 0, item.replace);        
             }
+        } catch (GLib.RegexError e) {
+            return "";
         }
 
+        if (line.contains ("---")) return "";
+        
+        // What is the purpose of this line?
```

Comment:
This is because `&` breaks if it's not "escaped" like `&amp;` on the label's markdown. Please delete this comment :)